### PR TITLE
ADEN-11569 extend galactus targeting with wiki and discussion interac…

### PR DESCRIPTION
**Pull request by mfarul**
…tions

### CODE REVIEW

File: src/ad-services/silver-surfer/silver-surfer-profile-fetcher.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { utils } from '@ad-engine/core';
 
 const logGroup = 'silver-surfer';
 
-class SilverSurferServiceLoader {
+export default class SilverSurferProfileFetcher {
```
Comment (nandy-andy) [inline src/ad-services/silver-surfer/silver-surfer-profile-fetcher.ts:5]: We may want to check if `export default` doesn't mess anything in how webpack generates bundles. It's a bit blurry for me and maybe @dmnsobczak or @preludek remember it better but I recall at some point somebody (Bielik or Jacek) were advocating for not using `export defaults` as it breaks tree-shaking or something 😊

Comment (mfarul) [reply src/ad-services/silver-surfer/silver-surfer-profile-fetcher.ts]: I've removed those defaults.

File: src/ad-services/silver-surfer/silver-surfer-profile-extender.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+enum InteractionType {
+	WIKI_CONTENT = 'wiki content',
+	DISCUSSIONS = 'discussions',
+}
+
+export class SilverSurferProfileExtender {
```
Comment (Tomasz-at-Fandom) [inline src/ad-services/silver-surfer/silver-surfer-profile-extender.ts:6]: I don't see any tests for this. Is this tested somewhere?

File: @types/silver-surfer/index.d.ts [hunk 3]
```diff
@@ -13,52 +12,59 @@ interface SilverSurferClient {
...
+type FlatAdTags = Partial<Record<AdTagKey, Array<string>>>;
 
-interface Page {
-	id: number;
-	communityId: number;
-	vertical: string;
-	adTags: AdTags;
-}
+type ProductName = 'mw' | 'dis';
+type MwVariant = 'gamepedia' | 'mobile-wiki' | 'oasis';
```
Comment (preludek) [inline @types/silver-surfer/index.d.ts:21]: Is this in the API? 🤔 Those are all legacy skins.

File: src/ad-services/silver-surfer/silver-surfer-profile-extender.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	extend(profile: UserProfile): UserProfile {
+		profile.interactions = this.extendInteractions(profile.interactions ?? []);
+		return profile;
+	}
+
+	getContext(): SilverSurferContext {
+		if (!window.SilverSurferSDK || !window.SilverSurferSDK.isInitialized()) {
+			console.warn('SilverSurfer SKD not available or not initialised!');
+			return { pages: [] } as SilverSurferContext;
```
Comment (Tomasz-at-Fandom) [inline src/ad-services/silver-surfer/silver-surfer-profile-extender.ts:15]: SilverSurferContext interface declares more none optional keys then just `pages`... this is cheating :P  Let's either do what TJ suggests [here](https://github.com/Wikia/ad-engine/pull/1294/files#r821697752), or make other keys optional or  make a SilverSurferContext class that will have some defaults.

File: @types/silver-surfer/index.d.ts (hunk 3 above)
Comment (mfarul) [reply @types/silver-surfer/index.d.ts:21]: yep, just copied it from SS, those are there probably due to data compatibility. I'll remove those, as we do not actually need this field.

File: src/ad-services/silver-surfer/silver-surfer-profile-extender.ts (hunk 2 above)
Comment (mfarul) [reply src/ad-services/silver-surfer/silver-surfer-profile-extender.ts:10]: yes, the test that is extended here covers both classes.

File: src/ad-services/silver-surfer/silver-surfer-profile-extender.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	DISCUSSIONS = 'discussions',
+}
+
+export class SilverSurferProfileExtender {
+	extend(profile: UserProfile): UserProfile {
+		profile.interactions = this.extendInteractions(profile.interactions ?? []);
+		return profile;
+	}
+
+	getContext(): SilverSurferContext {
```
[the remark this answers is not in the document]
Comment (mfarul) [reply src/ad-services/silver-surfer/silver-surfer-profile-extender.ts:16]: I've moved to have a minimum data version ;)

File: src/ad-services/silver-surfer/silver-surfer-profile-extender.ts (hunk 2 above)
Comment (Tomasz-at-Fandom) [reply src/ad-services/silver-surfer/silver-surfer-profile-extender.ts:10]: Are you sure? As far as I can see object instances are created, but then their methods are monkey patched with stubs.

File: src/ad-services/silver-surfer/silver-surfer-profile-extender.ts (hunk 4 above)
Comment (mfarul) [reply src/ad-services/silver-surfer/silver-surfer-profile-extender.ts]: I've moved to have a minimum data version ;)

File: src/ad-services/silver-surfer/silver-surfer-profile-extender.ts (hunk 2 above)
Comment (mfarul) [reply src/ad-services/silver-surfer/silver-surfer-profile-extender.ts:10]: yes,  the configureUserTargeting method is calling this. Just look on the new field interactions that is now required in the test to be filled with data based on pages.

Comment (Tomasz-at-Fandom) [reply src/ad-services/silver-surfer/silver-surfer-profile-extender.ts:10]: Oh sorry! I misread something. I thought you were stubbing the `SilverSurferProfileExtender.extend` method as well.

Btw the code coverage tool confirms that this file is in fact mostly covered :)
```
 src/ad-services/silver-surfer          |   78.04 |    55.81 |   73.91 |   77.33 |                                                                                                                                                                 
  index.ts                              |   95.23 |    73.33 |     100 |   97.22 | 56                                                                                                                                                              
  silver-surfer-profile-extender.ts     |   81.81 |    72.22 |   83.33 |   80.95 | 17-21                                                                                                                                                           
  silver-surfer-profile-fetcher.ts      |   33.33 |        0 |   16.66 |   33.33 | 9-17,21-41  
```

**Review by Tomasz-at-Fandom — APPROVED** [review]

**Pull request merged**